processing/src/main/java/org/apache/druid/query/rowsandcols/MapOfColumnsRowsAndColumns.java

```diff
@@ -80,6 +84,35 @@ public static MapOfColumnsRowsAndColumns fromMap(Map<String, ? extends Column> m
     );
   }
 
+  public static MapOfColumnsRowsAndColumns fromResultRow(ArrayList<ResultRow> objs, RowSignature signature)
```

Comment:
This change since it means we materialize everything in memory. Is it cool ?

Reply:
Not cool but with the 2 pass approach this needs a revisit which can be done as a follow up

Reply:
Lets keep this comment open as it would be helpful for follow on patches.